diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 57a74f5..6855b5b 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -9,7 +9,7 @@ endif()
 
 macro( hepmc_test testname )
   message( STATUS "HepMC3 test: building ${testname} for ${testname}" )
-  find_file( ${testname}_source ${testname}.cc ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR} )
+  find_file( ${testname}_source ${testname}.cc PATHS ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR} )
   ADD_EXECUTABLE(${testname} ${${testname}_source}   ${ARGN} )
   target_include_directories(${testname} BEFORE
   PUBLIC  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
@@ -31,7 +31,7 @@ endmacro( hepmc_test )
 
 macro( hepmc_unix_test testname )
   message( STATUS "HepMC3 test: building UNIX test ${testname} " )
-  find_file( ${testname}_source ${testname}.cc ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR} )
+  find_file( ${testname}_source ${testname}.cc PATHS ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR} )
   ADD_EXECUTABLE(${testname} ${${testname}_source}  ${ARGN} )
   target_include_directories(${testname} BEFORE
   PUBLIC  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
@@ -57,7 +57,7 @@ endmacro( hepmc_unix_test )
 
 macro( hepmc_root_test testname )
   message( STATUS "HepMC3 test: building ROOT test ${testname} " )
-  find_file( ${testname}_source ${testname}.cc ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR} )
+  find_file( ${testname}_source ${testname}.cc PATHS ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR} )
   ADD_EXECUTABLE(${testname} ${${testname}_source}  ${ARGN} )
   target_include_directories(${testname} BEFORE
   PUBLIC  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
@@ -81,7 +81,7 @@ endmacro( hepmc_root_test )
 
 macro( hepmc_protobuf_test testname )
   message( STATUS "HepMC3 test: building protobuf test ${testname} " )
-  find_file( ${testname}_source ${testname}.cc ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR} )
+  find_file( ${testname}_source ${testname}.cc PATHS ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR} )
   ADD_EXECUTABLE(${testname} ${${testname}_source}  ${ARGN} )
   target_include_directories(${testname} BEFORE
   PUBLIC  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
@@ -369,7 +369,7 @@ endif()
 
 macro( hepmc_convert_test testname )
   message( STATUS "HepMC3 test: building Convert test ${testname} with HEPMC2" )
-  find_file( ${testname}_source ${testname}.cc ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR} )
+  find_file( ${testname}_source ${testname}.cc PATHS ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR} )
   message( STATUS "HepMC3 test: building Convert test ${testname} ${${testname}_source} with HEPMC2" )
   ADD_EXECUTABLE(convert_${testname} ${${testname}_source}  ${ARGN} )
   target_include_directories(convert_${testname} BEFORE
